src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -3009,15 +3034,33 @@ static Value createObjectPropertyFunctionName(ExecutionState& state, const Value
 NEVER_INLINE void InterpreterSlowPath::createObjectPrepareOperation(ExecutionState& state, CreateObjectPrepare* code, ByteCodeBlock* byteCodeBlock, Value* registerFile)
 {
     if (code->m_stage == CreateObjectPrepare::Init) {
-        CreateObjectPrepare::CreateObjectData* data = new CreateObjectPrepare::CreateObjectData(code->m_propertyReserveSize, new Object(state));
-        registerFile[code->m_dataRegisterIndex] = Value(Value::FromPayload, reinterpret_cast<intptr_t>(data));
+        void* ptr;
+        if (byteCodeBlock->codeBlock()->isAsyncOrGenerator()) {
+            // async or generator uses ValueVector alloctor for allocating registerFile
+            // ValueVector allocator cannot track CreateObjectData if there is CreateObjectData on registerFile
+            ptr = GC_MALLOC(sizeof(CreateObjectPrepare::CreateObjectData));
+            registerFile[code->m_dataRegisterIndex] = Value(Value::FromPayload, reinterpret_cast<intptr_t>(ptr));
+        } else {
+            ptr = &registerFile[code->m_dataRegisterIndex];
+        }
+        CreateObjectPrepare::CreateObjectData* data = new (ptr) CreateObjectPrepare::CreateObjectData(
+            code->m_allPrecomputed, code->m_cachedObjectStructure.hasValue(), code->m_allPrecomputed,
+            code->m_propertyReserveSize, new Object(state), code);
```

Comment:
(Correct me if I'm wrong)
This part seems a bit confusing for me.
From this code, is `CreateObjectData` allocated in the registerFile except for async/generator functions?
If it is, it seems that the size of `CreateObjectData` exceeds the size of slot of registerFile (size of Value).
And this may incur incorrect operations because some part of `CreateObjectData` will be overwritten later.

Reply:
If I want to allocate CreateObjectData on stack, I reserve more space for this.
see https://github.com/Samsung/escargot/pull/1312/files#diff-fd687ed84ecd98cf82e9b39afa7722f0afff0ecb9454f0f1e36194b17d5da2f5
new code of ObjectExpression.h:69

Reply:
Oh I see :)